querydsl-codegen-utils/src/main/java/com/querydsl/codegen/utils/SimpleCompiler.java

```diff
@@ -47,36 +47,8 @@ protected static boolean isSureFireBooter(URLClassLoader cl) {
         return false;
     }
 
-    public static String getClassPath(URLClassLoader cl) {
-        try {
-            List<String> paths = new ArrayList<String>();
-            if (isSureFireBooter(cl)) {
-                // extract MANIFEST.MF Class-Path entry, since the Java Compiler doesn't handle
-                // manifest only jars in the classpath correctly
-                URL url = cl.findResource("META-INF/MANIFEST.MF");
-                Manifest manifest = new Manifest(url.openStream());
-                String classpath = manifest.getMainAttributes().getValue("Class-Path");
-                for (String entry : classpath.split(" ")) {
-                    URL entryUrl = new URL(entry);
-                    String decodedPath = URLDecoder.decode(entryUrl.getPath(), "UTF-8");
-                    paths.add(new File(decodedPath).getAbsolutePath());
-                }
-            } else {
-                ClassLoader c = cl;
-                while (c instanceof URLClassLoader) {
-                    for (URL url : ((URLClassLoader)c).getURLs()) {
-                        String decodedPath = URLDecoder.decode(url.getPath(), "UTF-8");
-                        paths.add(new File(decodedPath).getAbsolutePath());
-                    }
-                    c = c.getParent();
-                }
-            }
-            return paths.stream().collect(Collectors.joining(File.pathSeparator));
-        } catch (UnsupportedEncodingException e) {
-            throw new CodegenException(e);
-        } catch (IOException e) {
-            throw new CodegenException(e);
-        }
+    public static String getClassPath(ClassLoader cl) {
+        return new ClassGraph().overrideClassLoaders(cl).getClasspath();
```

Comment:
A key design decision in QueryDSL 5 was keeping the transitive dependencies to a bare minimum. Some folks in the issue tracker mentioned even that frameworks should impose no further dependencies at all.

I know this dependency is just used in code gen and won't leak into the users application in _most_ cases. That is, unless the wrong dependency scope is used for the apt dependencies or a module with a runtime dependency on codegen is used (like `Alias.*`).

Was it necessary to update this code for Java 11?

Reply:
Classloaders aren't `URLClassLoader`s anymore. 

Alternative would be to copy/paste from that library (or any other source), but updating is necessary.

Reply:
Oh well, the code relied on a dependency anyway, that we now got rid of 😄